app/src/main/java/com/nmakademija/nmaakademija/fragment/ScheduleFragment.java

```diff
@@ -71,22 +74,24 @@ public void onFailure(Call<List<ScheduleEvent>> call, Throwable t) {
     }
 
     private void setScheduleItems(List<ScheduleEvent> scheduleEvents) {
+        Collections.sort(scheduleEvents, new ScheduleEventComparator());
 
         ScheduleAdapter adapter = new ScheduleAdapter(getContext(), scheduleEvents);
         scheduleRecyclerView.setAdapter(adapter);
         SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd", Locale.US);
-        ArrayList<Date> scheduleDates = new ArrayList<>();
+        HashSet<Date> scheduleDates = new HashSet<>();
         List<ScheduleSectionsAdapter.Section> sections = new ArrayList<>();
         for (int i = 0; i < scheduleEvents.size(); i++) {
             Date eventDate = scheduleEvents.get(i).getDate();
-            Date date = new Date(eventDate.getYear(), eventDate.getMonth(), eventDate.getDate());
+            Date date = new Date(eventDate.getYear(), eventDate.getMonth(), eventDate.getDate()); // date same as eventDate, time = 00:00:00
```

Comment:
This `Date` constructor is deprecated. Can we throw it away ? We can use date string formatted with `dateFormat` instead of  `date`